[BUG] Type generation error messages for `@sap/cds-common-content`

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Nature of Your Project

TypeScript

### Current Behavior

Using `@sap/cds-common-content` now leads to typer error logs:

```
[ERROR] Derived singular 'Language' for your entity 'sap.common.Languages', already exists. The resulting types will be erronous. Consider using '@singular:'/ '@plural:' annotations in your model or move the offending declarations into different namespaces to resolve this collision.
[ERROR] Derived singular 'Country' for your entity 'sap.common.Countries', already exists. The resulting types will be erronous. Consider using '@singular:'/ '@plural:' annotations in your model or move the offending declarations into different namespaces to resolve this collision.
[ERROR] Derived singular 'Currency' for your entity 'sap.common.Currencies', already exists. The resulting types will be erronous. Consider using '@singular:'/ '@plural:' annotations in your model or move the offending declarations into different namespaces to resolve this collision.
```

### Expected Behavior

I'd expect entities delivered from SAP to not throw an error log message, when used with typer.

### Steps To Reproduce

- Use one of the above contents from cds-common-content
- Execute `npx @cap-js/cds-typer '*' --outputDirectory @cds-models`

### Environment

```markdown
- **OS**: MacOs
- **Node**: v20.11.0
- **npm**: 10.4.0
- **cds-typer**: 0.16.0
- **cds**: 7.6.2
- "@sap/cds-common-content": "^1.3.1"
```


### Repository Containing a Minimal Reproducible Example

_No response_

### Anything else?

_No response_

same issue on version **0.18.1**

@cap-js/audit-logging: 0.6.0
@cap-js/cds-typer: 0.18.1
@sap/cds: 7.7.2
@sap/cds-compiler: 4.5.0


```log
[ERROR] Derived singular 'Language' for your entity 'sap.common.Languages', already exists. The resulting types will be erronous. Consider using '@singular:'/ '@plural:' annotations in your model or move the offending declarations into different namespaces to resolve this collision.
[ERROR] Derived singular 'Country' for your entity 'sap.common.Countries', already exists. The resulting types will be erronous. Consider using '@singular:'/ '@plural:' annotations in your model or move the offending declarations into different namespaces to resolve this collision.
[ERROR] Derived singular 'Currency' for your entity 'sap.common.Currencies', already exists. The resulting types will be erronous. Consider using '@singular:'/ '@plural:' annotations in your model or move the offending declarations into different namespaces to resolve this collision.
[ERROR] Derived singular 'Timezone' for your entity 'sap.common.Timezones', already exists. The resulting types will be erronous. Consider using '@singular:'/ '@plural:' annotations in your model or move the offending declarations into different namespaces to resolve this collision.
[DEBUG] Printing aspect sap.common.CodeList
[DEBUG] Printing aspect sap.common.TextsAspect
[DEBUG] Printing aspect cuid
[DEBUG] Printing aspect managed
[DEBUG] Printing aspect temporal
[DEBUG] Printing type User:
{
  "kind": "type",
  "doc": "Canonical user ID",
  "@title": "{i18n>UserID}",
  "@description": "{i18n>UserID.Description}",
  "type": "cds.String",
  "length": 255,
  "@odata.draft.enabled": false
}
[DEBUG] Printing aspect extensible

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/).
 - Importing an enum into a service will now generate an alias to the original enum, instead of incorrectly duplicating the definition
 - Returning entities from actions/ functions and using them as parameters will now properly use the singular inflection instead of returning an array thereof
 - Aspects are now consistently named and called in their singular form
+- Only detect inflection clash if singular and plural share the same namespace. This also no longer reports `sap.common` as erroneous during type creation 
 
 ## Version 0.19.0 - 2024-03-28
 ### Added

--- a/lib/visitor.js
+++ b/lib/visitor.js
@@ -258,15 +258,13 @@ class Visitor {
         // entities. So to see if we would run into a naming collision, we forcefully take the last
         // part of the name, so "A.B" and "A.Bs" just become "B" and "Bs" to be compared.
         // FIXME: put this in a util function
-        //if (singular.split('.').at(-1) === plural.split('.').at(-1)) {
         if (plural.split('.').at(-1) === `${singular.split('.').at(-1)}_`) {
-            //plural += '_'
             this.logger.warning(
                 `Derived singular and plural forms for '${singular}' are the same. This usually happens when your CDS entities are named following singular flexion. Consider naming your entities in plural or providing '@singular:'/ '@plural:' annotations to have a clear distinction between the two. Plural form will be renamed to '${plural}' to avoid compilation errors within the output.`
             )
         }
-        // as types are not inflected, their singular will always clash and there is also no plural for them anyway
-        if (!isType(entity) && singular in this.csn.xtended.definitions) {
+        // as types are not inflected, their singular will always clash and there is also no plural for them anyway -> skip
+        if (!isType(entity) && `${ns.asNamespace()}.${singular}` in this.csn.xtended.definitions) {
             this.logger.error(
                 `Derived singular '${singular}' for your entity '${name}', already exists. The resulting types will be erronous. Consider using '@singular:'/ '@plural:' annotations in your model or move the offending declarations into different namespaces to resolve this collision.`
             )

